policy/release/git_branch/git_branch_test.rego

```diff
@@ -0,0 +1,118 @@
+package git_branch_test
+
+import rego.v1
+import data.lib
+
+rule_data := {
+  "git_branch": {
+    "allowed_branch_patterns": ["^refs/heads/main$", "^refs/heads/release-[23]$"]
+  }
+}
+
+test_deny_with_disallowed_branch {
+  input := {
+    "attestations": [{
+      "statement": {
+        "predicate": {
+          "buildConfig": {
+            "tasks": [{
+              "invocation": {
+                "environment": {
+                  "annotations": {
+                    "pipelinesascode.tekton.dev/source-branch": "refs/heads/feature-branch"
+                  }
+                }
+              }
+            }]
+          }
+        }
+      }
+    }]
+  }
+
+  data := {"rule_data": rule_data}
+
+  deny contains result
+  result.msg == "invalid branch"
+  result.details[1] == "refs/heads/feature-branch"
```

Comment:
This is okay, but there are some convention we want to follow for doing tests like this. (More details to follow.)

Reply:
Something like this:

```
expected := {"msg": "invalid branch", "details": ...}
lib.assert_equal(expected, deny) with input as mock_input with data.rule_data as mock_rule_data
```